hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/logic/SigsAndPayerKeyScreen.java

```diff
@@ -62,13 +106,70 @@ public ResponseCodeEnum applyTo(SwirldsTxnAccessor accessor) {
             speedometers.cycleSyncVerifications();
         }
 
+        final var sigMeta = accessor.getSigMeta();
+        if (sigMeta != null) {
+            sigMeta.replacePayerHollowKeyIfNeeded(accessor.getSigMap());
+        }
+
         if (hasActivePayerSig(accessor)) {
             txnCtx.payerSigIsKnownActive();
+
+            if (sigMeta != null && sigMeta.hasReplacedHollowKey()) {
+                accounts.get()
+                        .getForModify(EntityNum.fromAccountId(txnCtx.activePayer()))
+                        .setAccountKey(sigMeta.payerKey());
+                trackHollowAccountCompletion(txnCtx.activePayer(), sigMeta.payerKey());
+            }
+        }
+
+        final var ethTxExpansion = spanMapAccessor.getEthTxExpansion(accessor);
+        if (properties.isLazyCreationEnabled()
+                && ethTxExpansion != null
+                && ethTxExpansion.result().equals(OK)) {
+            final var ethTxSigs = spanMapAccessor.getEthTxSigsMeta(accessor);
+            final var callerNum = aliasManager.lookupIdBy(wrapUnsafely(ethTxSigs.address()));
+            if (callerNum != EntityNum.MISSING_NUM) {
+                var account = accounts.get().getForModify(callerNum);
+                if (account.getAccountKey() == EMPTY_KEY) {
```

Comment:
⚠️ I think we definitely need `if (EMPTY_KEY.equals(account.getAccountKey()))` since after restarting from a saved state, the `MerkleAccountState.deserialize()` method will have a new empty `KeyList`; not the `EMPTY_KEY` singleton.

Reply:
Fixed.